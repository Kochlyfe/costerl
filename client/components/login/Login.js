import React, { useState } from 'react';
import { View, Alert, Button, Text, TextInput, Keyboard } from 'react-native';
import ButtonApp from '../button/Button';
import Field from '../field/Field';
import DateInput from '../dateinput/DateInput';

import config from '../../config';

const Login = ({ form, postUser, currentUser }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleChange = (text, cb) => {
    console.log('text-->', text);
    cb(text);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('email-->', email);
    console.log('password-->', password);

    if (email === '' || password === '') {
      Alert.alert('Please insert in both fields');
    } else {
      postUser({
        email,
        password
      });
      // const res = postUser({})
      // console.log('currentUser after posting-->', currentUser);
      // if (typeof currentUser[0] === 'string') {
      //   Alert.alert('Could not create user');
      // }
      setEmail('');
      setPassword('');
    }
  };

  return (
    <>
      <TextInput
        placeholder='Type email'
        onChangeText={email => handleChange(email, setEmail)}
        defualtValue={email}
        value={email}
      />
      <TextInput
        placeholder='password'
        onChangeText={password => handleChange(password, setPassword)}
        defualtValue={password}
        value={password}
      />

      <Button
        onPress={(e) => {
          handleSubmit(e);
        }}
        title='Register'
      />

      <Button
        onPress={(e) => {
          handleSubmit(e);
        }}
        title='Login'
      />
    </>
  );
};

export default Login;