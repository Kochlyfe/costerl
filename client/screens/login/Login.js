import React from 'react';

import { View, Text, Alert, StyleSheet } from 'react-native';
import LoginForm from '../../components/LoginForm/LoginForm.js';

import Ui from '../../screens/ui/Ui';

const Login = ({ navigation, registerUser, currentUser, loginUser }) => {

  const loginRender = () => {
    return (
      <>
        <View>

          <LoginForm registerUser={registerUser} loginUser={loginUser} />

        </View>
      </>
    );
  };

  let tobeRendered;

  if (typeof currentUser[currentUser.length - 1] === 'string') {
    if (currentUser[currentUser.length - 1].includes('password')) {
      Alert.alert('Username or password is incorrect üñï');
    } else {
      Alert.alert('user already exists ‚ò†Ô∏è');
    }
    tobeRendered = false;
  } else if (!currentUser.length) {
    tobeRendered = false;
  } else {
    tobeRendered = true;
  }

  return (
    <View style={styles.container}>
      <Text style={styles.text}>LOGIN</Text>
      <View style={styles.loginFormBox}>
        {tobeRendered ? <Ui currentUser={currentUser} navigation={navigation} /> : loginRender()}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 10,
    paddingTop: 10,
  },
  text: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  loginFormBox: {},
});

export default Login;
