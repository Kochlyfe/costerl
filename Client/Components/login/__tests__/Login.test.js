import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Alert } from 'react-native';
import Login from '../Login';

describe('Login', () => {
  // test('render Login form', () => {
  //   const { debug } = render(<Login />);
  //   debug();
  // });

  test('snapshot', () => {
    const { toJSON } = render(<Login />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('displays input fields', () => {
    const { getByPlaceholderText } = render(<Login />);
    expect(getByPlaceholderText('Type email')).toBeTruthy();
    expect(getByPlaceholderText('password')).toBeTruthy();
  });

  test('displays buttons text', () => {
    const { getByText } = render(<Login />);
    expect(getByText('Login')).toBeTruthy();
    expect(getByText('Register')).toBeTruthy();
  });

  test('calls "handleSubmit" when clicking on submit button', () => {
    const mockOnClick = jest.fn();
    const { getByText, getByPlaceholderText } = render(<Login postUser={mockOnClick}></Login>);
    fireEvent.changeText(getByPlaceholderText('password'), 'mypassword');
    fireEvent.changeText(getByPlaceholderText('Type email'), 'john@doe.com');
    fireEvent(getByText('Register'), 'click', { preventDefault: jest.fn() });
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  test('returns error when password field is empty', () => {
    const mockOnClick = jest.fn();
    jest.spyOn(Alert, 'alert');

    const { getByText, getByPlaceholderText } = render(<Login postUser={mockOnClick}></Login>);
    fireEvent.changeText(getByPlaceholderText('password'), '');
    fireEvent.changeText(getByPlaceholderText('Type email'), 'john@doe.com');
    fireEvent(getByText('Register'), 'click', { preventDefault: jest.fn() });
    expect(Alert.alert).toHaveBeenCalled();
  });

  test('returns error when email field is empty', () => {
    const mockOnClick = jest.fn();
    jest.spyOn(Alert, 'alert');

    const { getByText, getByPlaceholderText } = render(<Login postUser={mockOnClick}></Login>);
    fireEvent.changeText(getByPlaceholderText('password'), 'test');
    fireEvent.changeText(getByPlaceholderText('Type email'), '');
    fireEvent(getByText('Register'), 'click', { preventDefault: jest.fn() });
    expect(Alert.alert).toHaveBeenCalled();
  });

  // handleSubmit fires alert if empty
});
