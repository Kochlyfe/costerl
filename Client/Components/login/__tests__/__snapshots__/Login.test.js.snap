// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot 1`] = `
Array [
  <TextInput
    allowFontScaling={true}
    defualtValue=""
    onChangeText={[Function]}
    placeholder="Type email"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
    value=""
  />,
  <TextInput
    allowFontScaling={true}
    defualtValue=""
    onChangeText={[Function]}
    placeholder="password"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
    value=""
  />,
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Register
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>,
]
`;
